lib/utils.js

```diff
@@ -21,41 +19,35 @@ export const isHeaderDefined = (headers, header) => {
  * the changelog event object emitted from the internal
  * cache registry.
  *
- * @param {Object} item A changelog event object
- *
- * @returns {Boolean}
+ * @param {object} item A changelog event object
+ * @returns {boolean}
  */
-
-export const hasManifestChange = item => {
+export const hasManifestChange = (item) => {
     const oldVersion = item.oldVal ? item.oldVal.version : '';
     const newVersion = item.newVal ? item.newVal.version : '';
     return oldVersion !== newVersion;
 };
 
-
 /**
- * Check if a value is a HttpIncoming object or not. If not, it
+ * Check if a value is a Podium HttpIncoming object or not. If not, it
  * assume the incoming value is a context
  *
- * @param {Object} incoming A object
- *
- * @returns {HttpIncoming}
- */
-export const validateIncoming = (incoming = {}) => (Object.prototype.toString.call(incoming) === '[object PodiumHttpIncoming]');
-
-/**
- * @typedef {import("@podium/utils").AssetCss | import("@podium/utils").AssetJs} Asset
+ * @param {object} incoming
+ * @returns {boolean}
  */
+export const validateIncoming = (incoming = {}) =>
+    Object.prototype.toString.call(incoming) === '[object PodiumHttpIncoming]';
 
 /**
  * Filter assets array based on scope.
  * If scope property is not present, asset will be included (backwards compatibility)
  * If scope property is set to "all", asset will be included.
  * If scope is set to "content" and asset scope property is set to "fallback", asset will not be included
  * If scope is set to "fallback" and asset scope property is set to "content", asset will not be included
+ * @template {import("@podium/utils").AssetCss | import("@podium/utils").AssetJs} T[]
```

Comment:
Replaced with generics so the type is kept untouched on the `Response` class itself